mastodon_archive/fix.py

```diff
@@ -45,8 +46,11 @@ def fix_boosts(args):
                 print(str(n) + " " + status["reblog"]["url"])
 
     if confirmed and n > 0:
+        backup_file = status_file + ".bak"
+        print("Backing up", status_file, "to", backup_file)
+        shutil.copy2(status_file, backup_file)
```

Comment:
Can't we use move, here, since we're going to write data to status_file?

Reply:
If we run the sub-command `fix-boosts` twice the backup will be overwritten! I can catch the exception and give a helpful message. Shall I?

Reply:
Now that I'm looking at the code again… We're using core.save, which will also move the file to a backup:
```python
    if os.path.isfile(file_name):
        os.replace(file_name, file_name + '~')
```
So now I'm thinking, perhaps we should just not have the backup_file code here?
And if the core.save code has a problem, then we should fix it there. What sort of exceptions are you expecting?

Reply:
I thought `shutil.copy2` would raise an exception if the file exists, but it does not.

I see your point, we can do it in core.save. But I think it is better to copy. If the file is moved as a backup and then immediately the operation is interrupted `Ctrl+C`, we have to:

1. do signal handling
2. deal with the confusion of a missing archive

Reply:
The chances are rare, but it is a backup after all. In my defence, at least Vim saves a "swapfile" as a copy and does not move the opened files. Does Emacs do it differently :see_no_evil: (just kidding).

So I suggest something like this in `core.save`

```diff
     if os.path.isfile(file_name):
-        os.replace(file_name, file_name + '~')
+        import shutil
+        backup_file = file_name + '~'
+        print("Backing up", file_name, "to", backup_file)
+        shutil.copy2(file_name, backup_file)
+
```

Reply:
Additionally, a prompt saying that backup may be overwritten if it exists.

Reply:
Sounds OK to me!

Reply:
Done in d2f2863...df52709